appsync/index.js

```diff
@@ -136,22 +136,73 @@ async function getDetailsForVehicle(userId, vehicleId, authToken) {
     accelerationScore: tripSummaryData.accelerationScore,
     totalScore: tripSummaryData.score,
     travelDistanceThisYear: 100,
-    fuelLeft: 999,
     parking: recentTrip.location,
     timeTraveled: tripSummaryData.TODO,
     trips: tripsForVehicle,
     lifeLitresPerHundredKm: tripsForVehicle,
     recentTrip: recentTrip,
-    parkedVehicle: parkedUserVehicles
+    parkedVehicle: parkedUserVehicles,
+    fuelLeft: fuelLeft.kmsLeft,
+    averagePer100Km: fuelLeft.averagePer100Km,
+    litresLeft: fuelLeft.litresLeft
   };
   return finalResult;
 }
 
+const calculateFuelLeft = (refillData, trips) => {
+  const refillTimeStamp = new Date(refillData[0].timestamp);
+  const refillLitres = parseFloat(refillData[0].litres);
+
+  const filteredTrips = trips.filter(trips => {
+    return refillTimeStamp <= new Date(trips.endTime);
+  });
+
+  const totalLitresUsed = filteredTrips.reduce((total, trip) => {
+    return total + parseFloat(trip.litres);
+  }, 0);
+
+  const totalDistance = filteredTrips.reduce((total, trip) => {
+    return total + parseFloat(trip.distance);
+  }, 0);
+
+  const litresLeftFunc = (refillLitres, totalLitresUsed) => {
```

Comment:
I don't see value in having this as a function. I would suggest `const litresLeft = Math.max(0, (refillLitres || 0) - (totalLitresUsed || 0));`